spaceship/lib/spaceship/connect_api/models/age_rating_declaration.rb

```diff
@@ -83,6 +93,32 @@ def self.type
         1 => true
       }
 
+      def self.map_deprecations_if_possible!(hash)
+        messages = []
+        errors = []
+
+        # Deprecated as of App Store Connect API 1.3
+        # If 'gamblingAndContests' found, set value for 'contests' and 'gambling'
+        if hash.key?('gamblingAndContests')
+          value = hash.delete('gamblingAndContests')
+
+          # gambling can be mapped without any issues because both 'gamblingAndContests' and 'gambling' are booleans
+          hash['gambling'] = value
+
+          # contests can only be mapped from false to 'NONE' (need to error if true)
+          if value == true
+            errors << "'gamblingAndContests' could not be mapped to 'contests' - 'contests' requires a value of 'NONE', 'INFREQUENT_OR_MILD', or 'FREQUENT_OR_INTENSE'"
+            hash['contests'] = value # setting to unchanged value
+          else
+            hash['contests'] = 'NONE'
+          end
+
+          messages << "Age Rating 'gamblingAndContests' has been deprecated and split into 'gambling' and 'contests'"
+        end
+
+        return hash, messages, errors
+      end
```

Comment:
```suggestion
      def map_deprecation_if_possible(attributes)
        attributes = attributes.dup
        messages = []
        errors = []

        value = attributes.delete('gamblingAndContests')
        return attributes, messages, errors if value.nil?

        messages << "Age Rating 'gamblingAndContests' has been deprecated and split into 'gambling' and 'contests'"

        if value == true
          errors << "'gamblingAndContests' could not be mapped to 'contests' - 'contests' requires a value of 'NONE', 'INFREQUENT_OR_MILD', or 'FREQUENT_OR_INTENSE'"
          attributes['gambling'] = value
          attributes['contests'] = value
        else
          attributes['gambling'] = value
          attributes['contests'] = 'NONE'
        end

        return attributes, messages, errors
      end
```

This is an example, but I think if you copy a given hash (with `Object#dup`), you don't need to name the method with `!`. `!` for a mapping method sounds a bit strange to me. And you can reduce nesting by using the guard idiom?

```
irb(main):001:0> a = { a: 1, b: 2, c: 3}
=> {:a=>1, :b=>2, :c=>3}
irb(main):002:0> b = a.dup
=> {:a=>1, :b=>2, :c=>3}
irb(main):003:0> b.delete(:a)
=> 1
irb(main):004:0> a
=> {:a=>1, :b=>2, :c=>3}
```